src/order/filter/basic.lean

```diff
@@ -1742,6 +1765,9 @@ lemma ne_bot.map (hf : ne_bot f) (m : α → β) : ne_bot (map m f) :=
 
 instance map_ne_bot [hf : ne_bot f] : ne_bot (f.map m) := hf.map m
 
+lemma map_neq_bot {α β : Type*} (f : α → β) {F : filter α} (hF : F ≠ ⊥) : map f F ≠ ⊥ :=
+λ h, hF (map_eq_bot_iff.mp h)
+
```

Comment:
This lemma doesn't feel like it should exist - everywhere in the file we use `ne_bot f` instead of `f ≠ ⊥`, and it seems odd to copy just one lemma across. How do other proofs avoid needing this?

Reply:
`ne_bot` was introduced as a class in order to use type class instance search to handle some easy proofs. That's a nice idea, but sometimes it simply makes things more convoluted (with lots of `haveI : f.ne_bot`). I don't see why we should remove the possibility to use the non-class version. But if you want to work on redoing proofs (without making them unreadable) then you can try.